src/domain/fiat/index.ts

```diff
@@ -4,8 +4,8 @@ import {
   NonIntegerError,
 } from "@domain/errors"
 
-export const toCents = (amount: number): UsdCents => {
-  return amount as UsdCents
+export const toCents = (amount: number | bigint): UsdCents => {
+  return Number(amount) as UsdCents
```

Comment:
if we pass a big int then this conversion is not safe

```
> Number(BigInt("19007199254740992"))
19007199254740990
> BigInt("19007199254740992")
19007199254740992n
```

Reply:
Oh this is probably a wider problem because we do this a lot in the new code (e.g. [here](https://github.com/GaloyMoney/galoy/pull/1237/files#diff-cbba23c1f04bbbb08b4e3b06a49efcc7be05fa0ebd30e2aa78739812a2e9b6b8R18))

Reply:
```
Of course, you should bear in mind that your BigInt value must be within [Number.MIN_SAFE_INTEGER, Number.MAX_SAFE_INTEGER]
```
if we are constraining the value then I am not sure if the use of BigInt is useful

Reply:
I think part of the reason too is to avoid having floats in our data

Reply:
Ran over this with @bodymindarts as well... his thinking is that this is ok as-is for now too